### ISSUE TITLE
Compiler error expected when using an async lambda expression for an Action argument

### ISSUE THREAD

**Issue by EmielJongerius-FF** [dotnet/roslyn#85733]
**Version Used**: 
dotnet 10.0.401

**Steps to Reproduce**:

```
void F(Action t)
{
    t();
}

// Compiles, but should throw error
F(async () => { });

var asyncLambda = async () => { };

// Throws error: this is OK
F(asyncLambda);
```

**Expected Behavior**:
In the code above the line `F(async () => { });` should throw a compiler error, just like the line `F(asyncLambda);` does.

**Actual Behavior**:
`F(async () => { });` does not throw a compiler error.

**Comment by MihuBot** [comment]
I'm a bot. Here are possible related and/or duplicate issues (I may be wrong):
- https://github.com/dotnet/roslyn/issues/73745
- https://github.com/dotnet/csharplang/issues/3842
- https://github.com/dotnet/runtime/issues/71700
- https://github.com/dotnet/roslyn/issues/83175

**Comment by CyrusNajmabadi** [comment]
So, afaict, this is by design.  

In the first case (`F(async () => { })`), the compiler infers the type for the lambda using the provided target type (`Action`).  Such a lambda is convertible to `Action` (after all, `async void` funcs are allowed) so this works out.

In the second case, when you have `var asyncLambda = async () => { };`, because there is no given type, we infer the type of that lambda (in this case `Func<Task>`.  That type is not convertible to `Action`, so this errors.

Note: if you were to write: 

```c#
Action asyncLambda = async () => { };
F(asyncLambda);
```

you will see it succeed.
